test/L1ERC20Bridge.t.sol

```diff
@@ -3,33 +3,159 @@ pragma solidity ^0.8.0;
 
 import {Test} from "forge-std/Test.sol";
 import {IGovernor} from "openzeppelin/governance/Governor.sol";
+import {WormholeRelayerBasicTest} from "wormhole-solidity-sdk/testing/WormholeRelayerTest.sol";
 
+import {L1Block} from "src/L1Block.sol";
 import {IERC20Mint} from "src/interfaces/IERC20Mint.sol";
 import {FakeERC20} from "src/FakeERC20.sol";
 import {L1ERC20Bridge} from "src/L1ERC20Bridge.sol";
+import {L2ERC20} from "src/L2ERC20.sol";
 import {Constants} from "test/Constants.sol";
 import {GovernorMock} from "test/mock/GovernorMock.sol";
 
-contract L1ERC20BridgeTest is Test, Constants {
-  IERC20Mint erc20;
+contract L1ERC20BridgeHarness is L1ERC20Bridge {
+  constructor(
+    address _token,
+    address _relayer,
+    address _governor,
+    uint16 _sourceId,
+    uint16 _targetId
+  ) L1ERC20Bridge(_token, _relayer, _governor, _sourceId, _targetId) {}
 
-  function setUp() public {
-    vm.createSelectFork(vm.rpcUrl("avalanche_fuji"));
-    erc20 = new FakeERC20("Hello", "WRLD");
+  function withdraw(address account, uint256 amount) public {
+    _withdraw(account, amount);
+  }
+
+  function receiveWithdrawalWormholeMessages(
+    bytes memory payload,
+    bytes[] memory additionalVaas,
+    bytes32 callerAddr,
+    uint16 sourceChain,
+    bytes32 deliveryHash
+  ) public {
+    _receiveWithdrawalWormholeMessages(
+      payload, additionalVaas, callerAddr, sourceChain, deliveryHash
+    );
+  }
+}
+
+contract L1ERC20BridgeTest is Constants, WormholeRelayerBasicTest {
+  L2ERC20 l2Erc20;
+  FakeERC20 fake;
+  L1ERC20Bridge bridge;
+
+  constructor() {
+    setTestnetForkChains(6, 5);
+  }
+
+  function setUpSource() public override {
+    fake = new FakeERC20("Hello", "WRLD");
+    IGovernor gov = new GovernorMock("Testington Dao", fake);
+    bridge =
+    new L1ERC20Bridge(address(fake), wormholeCoreFuji, address(gov), wormholeFujiId, wormholePolygonId);
+  }
+
+  function setUpTarget() public override {
+    L1Block l1Block = new L1Block();
+    l2Erc20 =
+    new L2ERC20( "Hello", "WRLD", wormholeCoreMumbai, address(l1Block), wormholePolygonId, wormholeFujiId);
+  }
+}
+
+contract Constructor is Test, Constants {
+  function testFork_CorrectlySetAllArgs(address l1Erc) public {
+    FakeERC20 fake = new FakeERC20("Hello", "WRLD");
+    IGovernor gov = new GovernorMock("Testington Dao", fake);
+    L1ERC20Bridge bridge =
+    new L1ERC20Bridge(address(l1Erc), wormholeCoreFuji, address(gov), wormholeFujiId, wormholePolygonId);
+    assertEq(address(bridge.L1_TOKEN()), l1Erc, "L1 token is not set correctly");
```

Comment:
This is a good question. We are only setting that parameter in the contract that is being tested. The other parameters are set in inherited contracts and tested in those contracts. My thoughts was that testing in both would be redundant.

Reply:
True, though it seems we don't have unit tests for WormholeReceiver or WormholeSender base classes. Maybe pull out an issue for figuring out how to add some tests for those, and otherwise close this.

Reply:
#25